File: moderate/endian.hs
import Data.Binary (decode)
import Data.Binary.Put (putWord16host, runPut)
import Data.Word (Word8)

littleEndian :: Bool
littleEndian = (decode . runPut $ putWord16host 42 :: Word8) == 42

main :: IO ()
main | littleEndian = putStrLn "LittleEndian"
     | otherwise    = putStrLn "BigEndian"

File: moderate/endian.go
package main

import (
	"fmt"
	"unsafe"
)

func main() {
	var i int32 = 1
	p := unsafe.Pointer(&i)
	if *(*byte)(p) == 1 {
		fmt.Println("LittleEndian")
	} else {
		fmt.Println("BigEndian")
	}
}
